src/inmanta/agent/forking_executor.py

```diff
@@ -265,6 +303,19 @@ def __init__(
         self.cli_log = cli_log
         self.session_gid = session_gid
 
+        self.executor_map: dict[executor.ExecutorId, MPExecutor] = {}
+        self.agent_map: dict[str, set[executor.ExecutorId]] = collections.defaultdict(set)
+
+        self._locks: inmanta.util.NamedLock = inmanta.util.NamedLock()
+
+    def __add_executor(self, theid: executor.ExecutorId, the_executor: MPExecutor) -> None:
```

Comment:
Why two underscores? Doesn't that have a special meaning?

Reply:
private (python mangles these)

```python
class X:
   def __init__(self):
      self.__xx = "a"
 
>>> a=X()
>>> dir(a)
['_X__xx', '__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getstate__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__']
```